config.go

```diff
@@ -145,3 +146,52 @@ func WithPublicEndpointFn(fn func(r *http.Request) bool) Option {
 		cfg.PublicEndpointFn = fn
 	})
 }
+
+// ResponseModifier is a function that allow modification to the response before
+// being passed to the underlying handler.
+type ResponseModifier func(ctx context.Context, w http.ResponseWriter)
+
+// WithResponseModifier adds a response modifier to the list of response modifiers.
+//
+// Each modifiers will be invoked before executing the underlying handler. So every
+// modification done by the modifiers will be passed to the underlying handler.
+func WithResponseModifier(modFunc ResponseModifier) Option {
+	return optionFunc(func(cfg *config) {
+		cfg.ResponseModifiers = append(cfg.ResponseModifiers, modFunc)
+	})
+}
+
+// ResponseModifierTraceIDResponseHeader is a response modifier that sets the trace id
+// into response header. Please note that even though the trace is not sampled, the trace
+// id will be still set into the response header. If this behavior is not desirable you
+// can create your own response modifier.
+func ResponseModifierTraceIDResponseHeader(opt ResponseModifierTraceIDResponseHeaderOption) ResponseModifier {
+	return func(ctx context.Context, w http.ResponseWriter) {
+		// set the response header key
+		headerKey := DefaultTraceResponseHeaderKey
+		if opt.HeaderKeyFunc != nil {
+			headerKey = opt.HeaderKeyFunc()
+			if len(headerKey) == 0 {
+				panic("header key must not be empty")
```

Comment:
We need to also add test for testing this behaviour

Reply:
Since we move to initialization, I think we can move `headerKey` as well like this:

```go
func ResponseModifierTraceIDResponseHeader(opt ResponseModifierTraceIDResponseHeaderOption) ResponseModifier {
	if err := opt.Validate(); err != nil {
		panic(err)
	}

	// moved headerKey here
	headerKey := DefaultTraceResponseHeaderKey
	if opt.HeaderKeyFunc != nil {
		headerKey = opt.HeaderKeyFunc()
	}

	return func(ctx context.Context, w http.ResponseWriter) {
		// remove headerKey

		...
	}
}
```

Reply:
Haha.., you are right, @ilhamsyahids! 😁

Ok, updated in https://github.com/riandyrn/otelchi/pull/57/commits/10e1b7346d415e4ad6c9ec03f8d2ef143ab86b7f. Please recheck.